pkg/webui/console/components/gateway-data-form/index.js

```diff
@@ -69,19 +81,74 @@ const validationSchema = Yup.object().shape({
   status_public: Yup.boolean().default(false),
   schedule_downlink_late: Yup.boolean().default(false),
   auto_update: Yup.boolean().default(false),
+  schedule_anytime_delay: Yup.string().required(sharedMessages.validateRequired),
 })
 
 @bind
 class GatewayDataForm extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    error: PropTypes.error,
+    formRef: PropTypes.shape({}),
+    initialValues: PropTypes.shape({}),
+    /** React reference to be passed to the form */
+    onSubmit: PropTypes.func.isRequired,
+    /** SubmitBar contents */
+    update: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    formRef: undefined,
+    update: false,
+    error: '',
+    initialValues: {},
+  }
+
+  state = {
+    shouldDisplayWarning: false,
+  }
+
   onSubmit(values, helpers) {
     const { onSubmit } = this.props
     const castedValues = validationSchema.cast(values)
 
     onSubmit(castedValues, helpers)
   }
 
+  decodeDelayValue(value) {
+    const duration = value.split(unitRegexp)[0]
+    const unit = value.split(duration)[1]
+
+    this.setState({ shouldDisplayWarning: this.isNotValidDuration(duration, unit) })
```

Comment:
While it can be done like this too, I think the cleaner approach is to hook into the `onChange` event of the Field component and to check the value there.

```js
@bind
onScheduleAnytimeDelayChange(value) {
  this.setState({ shouldDisplayWarning: this.isNotValidDuration(value) }) // also add the decoding to the check function then
}
```